views/js/tools/getPlatformInfo.js

```diff
@@ -0,0 +1,116 @@
+define([
+    'lodash',
+    'jquery',
+    'util/url'
+], function (_, $, url) {
+    'use strict';
+
+    /**
+     * Some default values
+     * @type {Object}
+     * @private
+     */
+    var defaultConfig = {
+        browserVersionAction: 'whichBrowser',
+        browserVersionController: 'CompatibilityChecker',
+        browserVersionExtension: 'taoClientDiagnostic'
+    };
+
+    /**
+     * Gets the URL of the platform tester
+     * @returns {String}
+     */
+    function getTesterUrl(window, action, controller, extension) {
+        var document = window.document;
+        var navigator = window.navigator;
+        var screen = window.screen;
+        var params = {};
+        var e = 0;
+        var f = 0;
+
+        // append the browser user agent
+        params.ua = navigator.userAgent;
+
+        // detect browser family
+        e |= window.ActiveXObject ? 1 : 0;
+        e |= window.opera ? 2 : 0;
+        e |= window.chrome ? 4 : 0;
+        e |= 'getBoxObjectFor' in document || 'mozInnerScreenX' in window ? 8 : 0;
+        e |= ('WebKitCSSMatrix' in window || 'WebKitPoint' in window || 'webkitStorageInfo' in window || 'webkitURL' in window) ? 16 : 0;
+        e |= (e & 16 && ({}.toString).toString().indexOf("\n") === -1) ? 32 : 0;
+        params.e = e;
+
+        // gather info about browser functionality
+        f |= 'sandbox' in document.createElement('iframe') ? 1 : 0;
+        f |= 'WebSocket' in window ? 2 : 0;
+        f |= window.Worker ? 4 : 0;
+        f |= window.applicationCache ? 8 : 0;
+        f |= window.history && window.history.pushState ? 16 : 0;
+        f |= document.documentElement.webkitRequestFullScreen ? 32 : 0;
+        f |= 'FileReader' in window ? 64 : 0;
+        params.f = f;
+
+        // append a unique ID
+        params.r = Math.random().toString(36).substring(7);
+
+        // get the screen size
+        params.w = screen.width;
+        params.h = screen.height;
+
+        return url.route(action, controller, extension, params);
+    }
+
+    /**
+     * Check the user browser and os
+     * @param {Window} window - Need an access to the window object
+     * @param {Object} config
+     * @param {String} config.browserVersionAction - The name of the action to call to get the browser checker
+     * @param {String} config.browserVersionController - The name of the controller to call to get the browser checker
+     * @param {String} config.browserVersionExtension - The name of the extension containing the controller to call to get the browser checker
+     * @returns {Promise}
+     */
+    return function getPlatformInfo(window, config) {
+        var testerUrl;
+
+        config = _.defaults((config || {}), defaultConfig);
+
+        testerUrl = getTesterUrl(
+            window,
+            config.browserVersionAction,
+            config.browserVersionController,
+            config.browserVersionExtension
+        );
+
+        return new Promise(function (resolve, reject) {
```

Comment:
This would imply changing the controller server-side, along with changing the class hierarchy to get the proper helper, etc. which I don't feel that confortable to do